### PR TITLE
MLDB-1834 cleaner err msg for SELECT {*}

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: server/dataset_context.cc [hunk 1]
```diff
@@ -129,6 +129,14 @@ doGetColumn(const Utf8String & tableName,
     throw HttpReturnException(400, "Cannot read column \"" + columnName.toUtf8String() + "\" with no dataset.");
 }
 
+GetAllColumnsOutput
+SqlExpressionMldbScope::
+doGetAllColumns(const Utf8String & tableName,
+                function<ColumnName (const ColumnName &)> keep)
+{
+    throw HttpReturnException(400, "Cannot get all columns with no dataset.");
```
Comment (jeremybarnes) [inline server/dataset_context.cc]: It should say "cannot use wildcards with no from clause".

Comment (steadtler) [inline server/dataset_context.cc]: Note that this scope is used in many contexts, some of which a dataset is present. Also, doGetAllColumns is used for all wildcards, e.g.: x.prefix*, not just "all columns"

Comment (jeremybarnes) [inline server/dataset_context.cc]: Also, the doGetColumn below should say "cannot read column "<columnName>" with no from clause"

**Comment by mailletf** [comment]
+1

**Pull request merged**